src/Rings/MPolyQuo.jl

```diff
@@ -28,10 +28,10 @@ function show(io::IO, Q::MPolyQuo)
   print(io, "Quotient of $(Q.R) by $(Q.I)")
 end
 
-gens(Q::MPolyQuo) = [Q(x) for x = gens(Q.R)]
-ngens(Q::MPolyQuo) = ngens(Q.R)
-gen(Q::MPolyQuo, i::Int) = Q(gen(Q.R, i))
-Base.getindex(Q::MPolyQuo, i::Int) = Q(Q.R[i])
+gens(Q::MPolyQuo) = [Q(x) for x = gens(Q.R)]::Vector{elem_type(Q)}
+ngens(Q::MPolyQuo) = ngens(Q.R)::Int
+gen(Q::MPolyQuo, i::Int) = Q(gen(Q.R, i))::elem_type(Q)
+Base.getindex(Q::MPolyQuo, i::Int) = Q(Q.R[i])::elem_type(Q)
```

Comment:
This is kind of a bandaid, and e.g. `base_ring` below is not type stable as it is. To fix this "properly", I think we'd need to add another type argument to `MPolyQuo` to record the type of `Q.R`

Reply:
These kind of assertions are all over the place in Singular.jl and AbstractAlgebra.jl. I believe that they are used in place of more type parameters.